pkg/utils/kubernetes/health/deployment.go

```diff
@@ -85,3 +85,26 @@ func CheckDeployment(deployment *appsv1.Deployment) error {
 
 	return nil
 }
+
+// IsDeploymentProgressing returns false if the Deployment has been fully rolled out. Otherwise, it returns true along
+// with a reason, why the Deployment is not considered to be fully rolled out.
+func IsDeploymentProgressing(deployment *appsv1.Deployment) (bool, string) {
```

Comment:
Actually, the additional checks should all be covered by checking the `Progressing` condition.

And we don't want to do `replicas != deployment.Status.AvailableReplicas` in `IsDeploymentProgressing` because it is already covered by `CheckDeployment`. 
So maybe in the API server case, we should just call `IsDeploymentProgressing` and `CheckDeployment` afterwards?

Reply:
Maybe this is an option, yeah 👍🏻